pkg/operator/apiserver/controller/workload/workload.go

```diff
@@ -287,9 +287,26 @@ func (c *Controller) updateOperatorStatus(ctx context.Context, previousStatus *o
 		desiredReplicas = *(workload.Spec.Replicas)
 	}
 
+	selector, err := metav1.LabelSelectorAsSelector(workload.Spec.Selector)
+	if err != nil {
+		return fmt.Errorf("failed to construct label selector: %v", err)
+	}
+	matchingPods, err := c.podsLister.Pods(c.targetNamespace).List(selector)
+	if err != nil {
+		return err
+	}
```

Comment:
Is it guaranteed that workload will always have a selector set?

Reply:
The workload is a deployment and it does not allow an empty (select all) selector. Nevertheless I have removed this logic.